programs/cleos/main.cpp

```diff
@@ -399,7 +423,13 @@ fc::variant push_transaction( signed_transaction& trx, const std::vector<public_
       }
    } else {
       if (!tx_return_packed) {
-        return fc::variant(trx);
+         if (tx_unpack_data) {
+            fc::variant unpacked_data_trx;
+            abi_serializer::to_variant(trx, unpacked_data_trx, abi_serializer_resolver, abi_serializer_max_time);
```

Comment:
Should add a `try-catch` around this and revert to `fc::variant(trx);`. Then I think you can just make that the behavior and remove the new `-u` option.

Reply:
Done